server/application/application.go

```diff
@@ -2013,6 +2010,43 @@ func (s *Server) ListResourceActions(ctx context.Context, q *application.Applica
 	return &application.ResourceActionsListResponse{Actions: actionsPtr}, nil
 }
 
+func (s *Server) getUnstructuredLiveResourceOrApp(ctx context.Context, rbacRequest string, q *application.ApplicationResourceRequest) (obj *unstructured.Unstructured, res *appv1.ResourceNode, app *appv1.Application, config *rest.Config, err error) {
+	var (
+		gvk       schema.GroupVersionKind
+		name      string
+		namespace string
+	)
+	if q.GetKind() == "Application" && q.GetGroup() == "argoproj.io" && q.GetName() == q.GetResourceName() {
+		name = q.GetName()
+		namespace = s.appNamespaceOrDefault(q.GetAppNamespace())
+		app, err = s.appLister.Applications(namespace).Get(name)
+		if err != nil {
+			return nil, nil, nil, nil, fmt.Errorf("error getting app by name: %w", err)
+		}
+		if err = s.enf.EnforceErr(ctx.Value("claims"), rbacpolicy.ResourceApplications, rbacRequest, app.RBACName(s.ns)); err != nil {
+			return nil, nil, nil, nil, err
+		}
+		config, err = s.getApplicationClusterConfig(ctx, app)
+		if err != nil {
+			return nil, nil, nil, nil, fmt.Errorf("error getting application cluster config: %w", err)
+		}
+		gvk = app.GroupVersionKind()
```

Comment:
Looks like the app informer doesn't always return a gvk.

What if instead of making a kubectl call to get the app we just convert the app that we've already fetched to an unstructured? i.e. `kube.ToUnstructured(app)`

Reply:
That's much nicer! (I don't have much experience with any Go libraries, so didn't know this was possible when I wrote this function 😞).

Reply:
No worries! I wasn't aware of it either until I noticed that the informer wasn't always... informing.  :-P

Reply:
Ok, done and squashed, @crenshaw-dev! Please re-test to verify.